### PR TITLE
chore(Ledger-Suite): FI-1502 move icp and icrc ledger suites

### PR DESCRIPTION
**Pull request by NikolasHai**
This MR proposes the following changes:

1. Move the ledger suite of icp to its designated subdirectory

2. Move the ledger suite for icrc1 to its designated subdirectory

This MR is part of an effort to restructe the subdirectory `rs/rosetta-api` into two logically separate subdirectories:
- `rs/rosetta-api`: Rosetta related code
- `rs/ledger_suite`: ICP and ICRC1 ledger related code.
This MR only moves code into a new directory and changes dependencies accordingly.

### CODE REVIEW

File: rs/ledger_suite/icp/rosetta-integration-tests/BUILD.bazel
Comment (mbjorkqvist) [inline rs/ledger_suite/icp/rosetta-integration-tests/BUILD.bazel:1]: Would the `rs/rosetta-api` directory be a better place for `rosetta-integration-tests`?

File: rs/ledger_suite/icrc1/rosetta/BUILD.bazel
Comment (mbjorkqvist) [inline rs/ledger_suite/icrc1/rosetta/BUILD.bazel:1]: Would this `rosetta` directory fit better under `rs/rosetta-api`? (I think I already made a similar comment somewhere but couldn't find it!)

**Comment by NikolasHai** [comment]
> Thanks a lot for taking on this task, @NikolasHai! This looks very good, although the sheer amount of changes is causing the github webui to be quite slow! I left comments mainly around the following:
> 
>     * Changes to archived proposal `.md` files from the cross-chain team should be reverted
> 
>     * Dependencies in `BUILD.bazel` files need to be re-sorted (I commented on a few only, but the CI `buildifier` job should anyway catch them)
> 
>     * A few unrelated rosetta-changes seem to have snuck in
> 
>     * Some guides that give an example git commit, plus a command related to a soon-to-be-changed path, will need to be changed in a follow-up PR once this PR has been merged and we know of a valid git commit that could be set in those guides
> 
>     * A couple of rosetta-related directories are now under `rs/ledger_suites`, is that on purpose? I didn't dig into it, but they would seem to fit better under `rs/rosetta-api`

Thanks for taking a look.
1. Thanks for catching that, I'll revert that
2. I'll run the buildifier, I believe sorting should be done with the tool you sent me the other day. WDYT? 
3. I wanted to base this branch on the one for neuron management change so I would not have to resolve the MR conflicts again once I merged that one, but I think it does not really matter. I will have MR conflicts no matter what. I will revert those changes. 
4. Do you know which guides those would be? 
5. Yes that is on purpose because I will move ICRC rosetta in a follow up MR. I wanted to make two, one for the ledgers and one for rosetta.

File: .github/workflows/rosetta-release.yml
Comment (NikolasHai) [reply .github/workflows/rosetta-release.yml:62]: I'll do that in the MR for moving rosetta and changing its dependencies.

File: ci/src/dependencies/scanner/manager/bazel_rust_dependency_manager_test.py
Comment (NikolasHai) [reply ci/src/dependencies/scanner/manager/bazel_rust_dependency_manager_test.py:1]: I would assume they are due to automatic reformatting. I did not actively touch them.

File: rs/ledger_suite/icp/rosetta-integration-tests/BUILD.bazel
Comment (NikolasHai) [reply rs/ledger_suite/icp/rosetta-integration-tests/BUILD.bazel:1]: Will move this one in the Rosetta related MR

File: rs/rosetta-api/Cargo.toml
Comment (NikolasHai) [reply rs/rosetta-api/Cargo.toml:81]: I believe those are from the MR for neuron management, I can change them back and redo them when the neuron managment MR goes through.

File: rs/ledger_suite/icrc1/ledger/README.adoc
Comment (NikolasHai) [reply rs/ledger_suite/icrc1/ledger/README.adoc:29]: The IC_Version is set dynamiccaly, no?

File: rs/ledger_suite/tests/sm-tests/Cargo.toml
Comment (NikolasHai) [reply rs/ledger_suite/tests/sm-tests/Cargo.toml:2]: Yes, I'll do it in another MR

File: rs/rosetta-api/BUILD.bazel
Comment (NikolasHai) [reply rs/rosetta-api/BUILD.bazel:187]: The snuck in because of the neuron management test.

File: rs/cross-chain/proposal-cli/src/canister/mod.rs
Comment (gregorydemay) [inline rs/cross-chain/proposal-cli/src/canister/mod.rs:49]: @mbjorkqvist @NikolasHai : One drawback here is that if one were to try to make a proposal for a commit that pre-dates this moving, then this will fail because the tool will look for a file `rs/ledger_suite/icrc1/archive/archive.did` which is actually at `rs/rosetta-api/icrc1/archive/archive.did`. Solving this would require a bit more work (one would need to determine if the target commit is before or after this move) but I have the impression that it's not worth it ( we usually make upgrade proposals for new versions, not commit that are a lot in the past). In the worst case, one would need to use this tool at the same commit one wants to upgrade the canister to. WDYT?

Comment (mbjorkqvist) [reply rs/cross-chain/proposal-cli/src/canister/mod.rs:49]: Good catch, thanks @gregorydemay! Another issue is with the git log command in the proposal cli that would for the next version(s) probably need to look in both places?

Comment (gregorydemay) [reply rs/cross-chain/proposal-cli/src/canister/mod.rs:49]: > Another issue is with the git log command in the proposal cli that would for the next version(s) probably need to look in both places?

yes, that's also a good point. Also here I have the impression that this would be a one-off issue so doing the proper solution here might not be worth it.

**Review by dist1ll — APPROVED** [review]

**Comment by mbjorkqvist** [comment]
> 2. I'll run the buildifier, I believe sorting should be done with the tool you sent me the other day. WDYT?

`bazel run //:buildifier` should take care of the `BUILD.bazel` files, and maybe you could run something like [this](https://crates.io/crates/cargo-sort) for the `Cargo.toml` files.

> 3. I wanted to base this branch on the one for neuron management change so I would not have to resolve the MR conflicts again once I merged that one, but I think it does not really matter. I will have MR conflicts no matter what. I will revert those changes.

Those changes would then probably not appear in this PR if you set the target branch to the one with the neuron management change, but currently this branch is still targeting `master`.

> 4. Do you know which guides those would be?

One is the one in `rs/ethereum/cketh/playground/README.adoc` (and I think I commented on another similar occurrence in this PR). In that file, there are the following instructions:

```
git checkout 5553ee5f7586290a59ed372a2b7aca847520af82
openssl sha256 bazel-bin/rs/ledger_suite/icrc1/ledger/ledger_canister_u256.wasm.gz
```

If the path for the `openssl` command is updated now, it won't make sense, since at the checkout of commit `5553ee5f7586290a59ed372a2b7aca847520af82` the path won't exist. I think the best way forward would be to change both the git commit ID, and the path in the `openssl` command, once this PR has been merged and the wasm archive is in the new location, WDYT?

**Comment by NikolasHai** [comment]
> > 2. I'll run the buildifier, I believe sorting should be done with the tool you sent me the other day. WDYT?
> 
> `bazel run //:buildifier` should take care of the `BUILD.bazel` files, and maybe you could run something like [this](https://crates.io/crates/cargo-sort) for the `Cargo.toml` files.
> 
> > 3. I wanted to base this branch on the one for neuron management change so I would not have to resolve the MR conflicts again once I merged that one, but I think it does not really matter. I will have MR conflicts no matter what. I will revert those changes.
> 
> Those changes would then probably not appear in this PR if you set the target branch to the one with the neuron management change, but currently this branch is still targeting `master`.
> 
> > 4. Do you know which guides those would be?
> 
> One is the one in `rs/ethereum/cketh/playground/README.adoc` (and I think I commented on another similar occurrence in this PR). In that file, there are the following instructions:
> 
> ```
> git checkout 5553ee5f7586290a59ed372a2b7aca847520af82
> openssl sha256 bazel-bin/rs/ledger_suite/icrc1/ledger/ledger_canister_u256.wasm.gz
> ```
> 
> If the path for the `openssl` command is updated now, it won't make sense, since at the checkout of commit `5553ee5f7586290a59ed372a2b7aca847520af82` the path won't exist. I think the best way forward would be to change both the git commit ID, and the path in the `openssl` command, once this PR has been merged and the wasm archive is in the new location, WDYT?

Sounds good to me. I' create an MR for it.

File: rs/ledger_suite/tests/sm-tests/Cargo.toml
Comment (NikolasHai) [reply rs/ledger_suite/tests/sm-tests/Cargo.toml:1]: I created a ticket for it

File: rs/ledger_suite/icrc1/ledger/README.adoc
Comment (NikolasHai) [reply rs/ledger_suite/icrc1/ledger/README.adoc:29]: I created a ticket for it.

**Review by cgundy — APPROVED** [review]

File: rs/cross-chain/proposal-cli/src/canister/mod.rs
Comment (NikolasHai) [reply rs/cross-chain/proposal-cli/src/canister/mod.rs:49]: What are that odds that we try to upgrade to a commit that predates the commit after this merge?

**Review by schneiderstefan — APPROVED** [review]

File: rs/ledger_suite/common/ledger_canister_core/BUILD.bazel
Comment (alin-at-dfinity) [inline rs/ledger_suite/common/ledger_canister_core/BUILD.bazel:14]: [Adding this here, because GitHub only forces you to resolve inline comments, not top-level ones.]

You should make @dfinity/finint an owner of `rs/ledger_suite` in `.github/CODEOWNERS`. As is, it's @dfinity/ic-interface-owners that owns it (along with everything not explicitly owned under `rs`).

**Review by alin-at-dfinity — APPROVED** [review]

File: rs/ledger_suite/common/ledger_canister_core/BUILD.bazel
Comment (NikolasHai) [reply rs/ledger_suite/common/ledger_canister_core/BUILD.bazel:14]: Thanks, I edited the codeowner part.

**Review by mbjorkqvist — APPROVED** [review]
Thanks for addressing the comments, @NikolasHai!

**Comment by NikolasHai** [comment]
> @NikolasHai I think you can revert the changes under `ci/src/dependencies`. This is just some test data that was generated out of an earlier version of the repo. It must not be kept up-to-date.

Done

**Review by mraszyk — APPROVED** [review]

File: rs/cross-chain/proposal-cli/src/canister/mod.rs
Comment (gregorydemay) [reply rs/cross-chain/proposal-cli/src/canister/mod.rs:49]: > What are the odds that we try to upgrade to a commit that predates the commit after this merge?

I think this is unlikely, but it will impact the next round of proposals for all canisters supported by this tool, which are just not XC canisters but also FI canisters, that's why I wanted to raise the issue. But I agree that this should be a one-off problem and so I don't think we need anything fancy.

**Review by gregorydemay — APPROVED** [review]

**Pull request merged**
